### ISSUE TITLE
Getting some error while downloading any file from Dropbox

### ISSUE THREAD

**Issue by iChirag** [dropbox/dropbox-sdk-obj-c#180]
> DropboxClientError[{
>     NSError = "Error Domain=NSCocoaErrorDomain Code=4 \"\U201cCFNetworkDownload_PqQdEJ.tmp\U201d couldn\U2019t be moved to \U201c123\U201d because either the former doesn\U2019t exist, or the folder containing the latter doesn\U2019t exist.\" UserInfo={NSSourceFilePathErrorKey=/Users/ichirag/Library/Developer/CoreSimulator/Devices/3D50FBB7-3A3F-4F21-9BA2-69765A1837CC/data/Containers/Data/Application/53D22640-B18B-4D50-B541-2EA2E8621979/Library/Caches/com.apple.nsurlsessiond/Downloads/com.sn.shouse.pro/CFNetworkDownload_PqQdEJ.tmp, NSUserStringVariant=(\n    Move\n), NSDestinationFilePath=/Users/ichirag/Library/Developer/CoreSimulator/Devices/3D50FBB7-3A3F-4F21-9BA2-69765A1837CC/data/Containers/Data/Application/53D22640-B18B-4D50-B541-2EA2E8621979/Documents/Gallery/Docs/123/iaccounts.xlsx, NSFilePath=/Users/ichirag/Library/Developer/CoreSimulator/Devices/3D50FBB7-3A3F-4F21-9BA2-69765A1837CC/data/Containers/Data/Application/53D22640-B18B-4D50-B541-2EA2E8621979/Library/Caches/com.apple.nsurlsessiond/Downloads/com.sn.shouse.pro/CFNetworkDownload_PqQdEJ.tmp, NSUnderlyingError=0x600000a54df0 {Error Domain=NSPOSIXErrorDomain Code=2 \"No such file or directory\"}}";
> }];

here is my code:

```
-(void)downloadFileFromDropBox:(NSString *)filePath {
//    [self.restClient loadFile:filePath intoPath:[self.strDownloadPath stringByAppendingPathComponent:[filePath lastPathComponent]]];
    
    [Utils showActivity];
    DBUserClient *client = [DBClientsManager authorizedClient];

    NSString *urlPath = [self.strDownloadPath stringByAppendingPathComponent:[filePath lastPathComponent]];
   
    [[[client.filesRoutes downloadUrl:filePath overwrite:YES destination:[NSURL fileURLWithPath:urlPath isDirectory:NO]]
      setResponseBlock:^(DBFILESFileMetadata *result, DBFILESDownloadError *routeError, DBRequestError *networkError,
                         NSURL *destination) {
          if (result) {
              NSLog(@"%@\n", result);
              NSData *data = [[NSFileManager defaultManager] contentsAtPath:[destination path]];
              NSString *dataStr = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
              NSLog(@"%@\n", dataStr);
              
              
              dispatch_async(dispatch_get_main_queue(), ^{
                  
                  [Utils alertMessageTitle:@"" message:@"File downloaded successfully." CompletionHandler:nil];
                  [self dismissViewControllerAnimated:YES completion:nil];
                  [Utils dismissActivity];
                  
              });
              
          } else {
              NSLog(@"%@\n%@\n", routeError, networkError);

              [Utils dismissActivity];
          }
      }] setProgressBlock:^(int64_t bytesDownloaded, int64_t totalBytesDownloaded, int64_t totalBytesExpectedToDownload) {
          NSLog(@"%lld\n%lld\n%lld\n", bytesDownloaded, totalBytesDownloaded, totalBytesExpectedToDownload);
      }];
    
}

```

**Comment by greg-db** [comment]
@iChirag How is your `<MY VALID PATH>` built?

**Comment by iChirag** [comment]
It is a valid path. See this I get using `NSlog`

`/Users/myname/Library/Developer/CoreSimulator/Devices/3D50FBB7-3A3F-4F21-9BA2-69765A1837CC/data/Containers/Data/Application/1EF1BB7D-EEB7-457E-B97B-FE744875A516/Documents/Gallery/Docs/123/DNO_03.jpg`

**Comment by greg-db** [comment]
@iChirag Can you share the code the produces the `/Users/myname/Library/Developer/CoreSimulator/Devices...` path? That will likely be relevant in reproducing this issue. Thanks in advance!

**Comment by iChirag** [comment]
okay here is code line by which that path produced

`NSString *urlPath = [self.strDownloadPath stringByAppendingPathComponent:[filePath lastPathComponent]];`

This is called inside the function.

That function is called by using following code in **didSelectRowAtIndexPath**

```
DBFILESFileMetadata *fileMetadata = (DBFILESFileMetadata *)entry;
NSLog(@"File data: %@\n", fileMetadata);
[self performSelector:@selector(downloadFileFromDropBox:) withObject:fileMetadata.pathLower afterDelay:.1];

```

**Comment by greg-db** [comment]
Apologies, I should have asked earlier, but what version of the SDK do you have installed? If it's not the latest version, currently v3.3.4, please update to that and let us know if that helps. There have been a number of bug fixes in recent versions.

Anyway, if that doesn't help, we can look into it, but we'll still need a bit more information to try to reproduce this. Specifically, the code you shared doesn't show how you're building `self.strDownloadPath`. Please share that to help us investigate. Thanks!

**Comment by iChirag** [comment]
okay It seems to be resolved with **v3.3.4**
However when I am updating with pod it taking some older version. So that was a mistake. Anyway,`pod 'ObjectiveDropboxOfficial','3.3.4'` resolved my issued.
